src/index.ts

```diff
@@ -2803,6 +2836,46 @@ class Observable<T> {
     }
 }
 
+/**
+ * HttpApiClient class.
+ * @ignore
+ * @internal
+ */
+class HttpApiClient {
+    private readonly apiUrl: string
+
+    constructor(apiUrl: string) {
+        this.apiUrl = apiUrl
+    }
+
+    doRequest<T>(request: HttpRequest<T>): Promise<T> {
+        return new Promise((resolve, reject) => {
+            const options = {
+                method: 'POST',
+                headers: {
+                    'Content-Type': 'application/json',
+                    'User-Agent': 'quadcode-client-sdk-js/0.1.3'
+                },
+                body: JSON.stringify(request.messageBody())
+            }
+
+            fetch(`${this.apiUrl}${request.path()}`, options)
+                .then((response) => {
+                    if (!response.ok) {
```

Comment:
Should give ability to user for checking http status code in user's code. Maybe should wrap response data to `Response` class with two properties `data` and `statusCode`.